test/MainDeployment.t.sol

```diff
@@ -0,0 +1,475 @@
+
+        // TODO: (4) Exclude v2 from fees on v1
+        gogeToken_v1.excludeFromFees(address(gogeToken_v2), true);
+
+        // TODO: (5) Perform migration -> 6 days
+        migrateActor(tim);
+        migrateActor(joe);
+        migrateActor(sal);
+        migrateActor(nik);
+        migrateActor(jon);
+
+        // Show price of v2
+        price = getPrice(address(gogeToken_v2));
+        emit log_named_uint("cost of 1 v2 token", price); // 0.000002119865796663
+
+        // TODO: (6) Perform mass airdrop to private sale contributors
+        // NOTE: IF USING BULKSENDER -> MAKE SURE TO WHITELIST BULKSENDER CONTRACT
+        gogeToken_v2.transfer(address(567), 20_000_000_000 ether);
+
+        // TODO: (7) enableTrading() on v2
+        gogeToken_v2.enableTrading();
+    }
+
+    // ~~ Utility Functions ~~
+
+    /// @notice Returns the price of 1 token in USD
+    function getPrice(address token) internal returns (uint256) {
+        address[] memory path = new address[](3);
+
+        path[0] = token;
+        path[1] = WBNB;
+        path[2] = BUSD;
+
+        uint256[] memory amounts = IUniswapV2Router01(UNIV2_ROUTER).getAmountsOut(1 ether, path);
+
+        return amounts[2];
+    }
+
+    /// @notice Creates v1 token holders. The holder balances should total just under 22B tokens
+    function createHolders() internal {
+        //Initialize wallet amounts.
+        uint256 amountJoe = 10_056_322_590 ether;
+        uint256 amountSal = 8_610_217_752 ether;
+        uint256 amountNik = 900_261_463 ether;
+        uint256 amountJon = 200_984_357 ether;
+        uint256 amountTim = 600_000 ether;
+
+        // Transfer tokens to Joe so he can migrate.
+        gogeToken_v1.transfer(address(joe), amountJoe);
+        gogeToken_v1.transfer(address(sal), amountSal);
+        gogeToken_v1.transfer(address(nik), amountNik);
+        gogeToken_v1.transfer(address(jon), amountJon);
+        gogeToken_v1.transfer(address(tim), amountTim);
+
+        // Verify amount v1 and 0 v2 tokens.
+        assertEq(gogeToken_v1.balanceOf(address(joe)), amountJoe);
+        assertEq(gogeToken_v1.balanceOf(address(sal)), amountSal);
+        assertEq(gogeToken_v1.balanceOf(address(nik)), amountNik);
+        assertEq(gogeToken_v1.balanceOf(address(jon)), amountJon);
+        assertEq(gogeToken_v1.balanceOf(address(tim)), amountTim);
+    }
+
+    /// @notice migrate tokens from v1 to v2
+    function migrateActor(Actor actor) internal {
+        uint256 bal = gogeToken_v1.balanceOf(address(actor));
+
+        // Approve and migrate
+        assert(actor.try_approveToken(address(gogeToken_v1), address(gogeToken_v2), gogeToken_v1.balanceOf(address(actor))));
+        assert(actor.try_migrate(address(gogeToken_v2)));
+
+        assertEq(gogeToken_v1.balanceOf(address(actor)), 0);
+        assertEq(gogeToken_v2.balanceOf(address(actor)), bal);
+    }
+
+    /// @notice Perform a buy
+    function buy_generateFees(uint256 tradeAmt) public {
```

Comment:
Are these buy and sell functions actually tests? Is the idea to perform a buy/sell with a fuzzed amount and confirm that the buy/sell goes through?

Reply:
It's a utility function that I call to perform a buy or sell at any moment of specified amounts to ensure correct taxation. Check `test_mainDeployment_feeDistributions_dev()` I do have a fuzzing test on buys/sells -> `test_royaltyTesting_generateFees_fuzzing` in Royalties.t.sol